sched/semaphore/sem_holder.c

```diff
@@ -103,20 +103,18 @@ nxsem_allocholder(FAR sem_t *sem, FAR struct tcb_s *htcb)
     {
       serr("ERROR: Insufficient pre-allocated holders\n");
       pholder          = NULL;
-      DEBUGPANIC();
     }
 
-  if (pholder != NULL)
-    {
-      pholder->sem    = sem;
-      pholder->htcb   = htcb;
-      pholder->counts = 0;
+  DEBUGASSERT(pholder != NULL);
```

Comment:
If we can run out of holder 2 ways here. Why leave this debug detection only? The following code will corrupt memory. Not having enough holders, if it were survivable will still cause PI issues,

Reply:
I just didn't want to change existing behaviour that use `DEBUGPANIC();` and not `PANIC();`. But I agree that `ASSERT(pholder != NULL);` might be better here.

Reply:
Done